sale_timesheet_order_line_no_update/models/project.py

```diff
@@ -0,0 +1,59 @@
+import json
+
+from odoo import api, fields, models
+
+
+class ProjectTask(models.Model):
+    _inherit = "project.task"
+
+    new_sale_line_id = fields.Many2one(
+        "sale.order.line",
+        "Default Sale Order Item",
+        copy=False,
+    )
+    hide_original_sol = fields.Boolean()
+
+    new_sale_line_id_domain = fields.Char(related="project_id.sale_line_id_domain")
+
+
+class ProjectProject(models.Model):
+    _inherit = "project.project"
+
+    select_all_project_sale_items = fields.Boolean()
+    sale_line_id_domain = fields.Char(
+        compute="_compute_sale_line_id_domain",
+        readonly=True,
+        store=False,
+    )
+
+    @api.depends("select_all_project_sale_items")
+    def _compute_sale_line_id_domain(self):
+        for rec in self:
+            if not self.select_all_project_sale_items:
```

Comment:
Why do you use `self.select_all_project_sale_items`?
May  be `rec.select_all_project_sale_items`?